src/controller.cpp

```diff
@@ -30,27 +31,30 @@ void Controller::configure(
   path_handler_.initialize(parent_, name_, costmap_ros_, tf_buffer_, parameters_handler_.get());
   trajectory_visualizer_.on_configure(parent_, costmap_ros_->getGlobalFrameID());
 
-  RCLCPP_INFO(logger_, "Configured MPPI Controller: %s", name_.c_str());
+  RCLCPP_INFO_STREAM(logger_, fmt::format("Configured MPPI Controller: {}", name_));
```

Comment:
fmt lib is in c++20 actually. So this just workaround that we don't have c++20 here yet, it's easy to use, and it was very helpful with parameters logging. Why not use it ? People anyways would learn new standards

Reply:
How parameters and especially dynamic parameters managed was not clean. A lot of dry principles violated. So basically what i did is provide easy to use api for better extensibility. Sometimes templates really needed. If constexpr stuff just a compile time check, it's not tepmlated itself

Reply:
Right now, Humble is targeting C++17, I think we're still a number of years away from having C++20 in ROS 2 due to the slow updates so that the hard real time certification of the standard library updates can be made before ROS 2 forces that upgrade. That restricted Nav2 to C++17 as well (or people get pretty worked up) but if its available as a separate library, from a versioning standpoint, that's OK.

Mostly I want this to be consistent with other code in Nav2, which has analogous logging to this in the different lifecycle transitions / elsewhere where logging is required. There's utility in consistency so that someone can take a look at any package within the project and see analogous tools being used everywhere so that there aren't some individual packages doing something very differently from everywhere else where analogs are had. Its the same rationale for making sure that if we do `<some operation>` in Nav2, I make sure all such instances of that operation are updated accordingly.

Reply:
Yeah, sure consistency is a nice stuff. 
fmt lib just add some nice features to format strings. that's not like it's changing the way we logging.
But i could easily revert this dependecy if you insist.

Reply:
I'm less sensitive to that than I am on the templating topic below -- but yes, I would appreciate if we kept the logging consistent across the board.

Reply:
That make sense)